vars/dockerBuildArgs.groovy

```diff
@@ -49,26 +101,43 @@ String call(Map config = [:]) {
       ret_str += " --build-arg CB0=" + current_time.get(Calendar.WEEK_OF_YEAR)
     }
 
-    if (add_repos) {
-      if (env.REPOSITORY_URL) {
-        ret_str += ' --build-arg REPO_URL=' + env.REPOSITORY_URL
-      }
-      if (env.DAOS_STACK_EL_7_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_EL7=' + env.DAOS_STACK_EL_7_LOCAL_REPO
+    String repo_base = env.REPOSITORY_URL
+    // No repo_base, no repos to add.
+    if (add_repos && repo_base) {
+      String repo_name = null
+      String repo_arg = ''
+      if (stage_info['target'] == 'centos7') {
+        repo_alias = 'EL_7'
+        repo_arg = 'EL7'
       }
-      if (env.DAOS_STACK_EL_8_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_EL8=' + env.DAOS_STACK_EL_8_LOCAL_REPO
+      if (stage_info['target'] == 'centos8') {
```

Comment:
Should use:
```
} else if (stage_info['target'] == 'centos8') {
```
here, and on all of the others below to prevent evaluating every branch when the matching one has already been found.

Reply:
Missed that on the refactor.